pylearn2/datasets/dense_design_matrix.py

```diff
@@ -786,7 +787,10 @@ def set_topological_view(self, V, axes=('b', 0, 1, 'c')):
                 dim = 1
             else:
                 dim = self.y.shape[-1]
-            # This is to support old pickled models
+
+            # This is to support old pickled models as well to trigger the
+            # conversion of the labels vector into a 2D one-hot matrix if
+            # y_labels is specified
```

Comment:
Ok, I just thought that it might be useful to know that it is not only to support old models, to avoid a "support dropped" situation as in the hdf5 equivalent.